hedera-node/src/main/java/com/hedera/services/files/SpecialFileSystem.java

```diff
@@ -0,0 +1,229 @@
+package com.hedera.services.files;
+
+import com.hedera.services.ledger.HederaLedger;
+import com.hedera.services.legacy.logic.ApplicationConstants;
+import com.hedera.services.state.merkle.MerkleToken;
+import com.hedera.services.utils.EntityIdUtils;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.FileID;
+import com.swirlds.common.FastCopyable;
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+import com.swirlds.common.merkle.MerkleLeaf;
+import com.swirlds.common.merkle.utility.AbstractMerkleNode;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.hedera.services.legacy.logic.ApplicationConstants.SPECIAL_FILESYSTEM_DIR;
+import static com.swirlds.common.CommonUtils.hex;
+
+/**
+ * Save some special system files in local file system instead of database to improve access efficiency
+ * File contents locate on local file system, but file hashes save in memory using a hash map
+ */
+public class SpecialFileSystem extends AbstractMerkleNode implements MerkleLeaf, FastCopyable<SpecialFileSystem> {
```

Comment:
To match existing naming conventions and clarify the purpose of the type, `MerkleDiskFs` seems a better name. Addressed in #658.